src/internal/direntry.rs

```diff
@@ -142,9 +142,10 @@ impl DirEntry {
         // Spec say this is suppose to be zero for DirEntries
         // but some cfb implementations set start_sector to FREE_SECTOR instead
         if obj_type == consts::OBJ_TYPE_STORAGE
-            && !(start_sector == 0 || start_sector == consts::FREE_SECTOR)
+            && !(start_sector == 0 || start_sector == consts::FREE_SECTOR
+            || start_sector == (consts::FREE_SECTOR - 1))
```

Comment:
`0xfffffffe` is `consts::END_OF_CHAIN`, so it would be better to use that rather than `(consts::FREE_SECTOR - 1)`.